### PR TITLE
console: Add Azure & Keycloak docs

### LINKED ISSUE #1481: Add Azure AD and Keycloak instructions for RP Console doc
### Describe the Issue

For the upcoming Redpanda Console release, we need content that describes how to create the OIDC application in Azure AD and Keycloak. This will supplement content provided in [PR 1429](https://github.com/redpanda-data/documentation/pull/1429).

### Updates to existing documentation

URL of the page(s) to which the updates apply:
https://docs.redpanda.com/docs/manage/security/console/
PR 1429 (submitted by Martin S.) adds a child page for "Azure" and one for "Keycloak" in this section.

### Helpful resources

Link to Azure recording on 3-30-23 (from Google Meet):
https://drive.google.com/file/d/1xhqy00dxJx1iY1AKMLTSBfkPDONX5aa3/view

Link to Keycloak recording on 4-6-23 (from Google Meet):
https://drive.google.com/file/d/1VAeAn5ADvogXRbgur_MWbmkC-u7I4Uh6/view

### If new feature, in which release is it included or expected? 

The next release of Redpanda Console will include support for Azure AD and Keycloak IdPs.

### Additional notes

 - Relevant GitHub issues and pull requests: [PR 1429](https://github.com/redpanda-data/documentation/pull/1429)
    - This is the existing PR submitted by Martin S. Information on configuring the two IdPs will be added to this PR.
 - Specific Redpanda contributors to notify: @weeco

### PR DESCRIPTION
**Pull request by weeco**
This PR adds Azure AD & Keycloak documentation for the upcoming Console release. It's still missing content that describes how to create the OIDC application in the respective identity provider. I asked Anne to help here and I'll meet with her next week to show how it's done. 

Resolves #1481

### CODE REVIEW

File: docs/manage/security/console/authentication.mdx [hunk 1]
```diff
@@ -3,21 +3,30 @@ title: Authentication
...
+
+- [AzureAD](../azure-ad)
 - [Google](../google)
 - [GitHub](../github)
+- [Keycloak](../keycloak)
 - [Okta](../okta)
 - [Generic OIDC](../generic-oidc)
 
-You can use one or more login providers at the same time. To enable SSO authentication 
+You can use one or more login providers at the same time. To enable SSO authentication
```
Comment (AnneLeeperRP) [inline docs/manage/security/console/authentication.mdx:29]: Add comma after introductory phrase:
```suggestion
You can use one or more login providers at the same time. To enable SSO authentication,
```

File: docs/manage/security/console/authentication.mdx [hunk 2]
```diff
@@ -28,18 +37,18 @@ to add Google login support looks like this:
...
-  # characters. For additional security, use a different secret key for 
-  # each environment. jwtSecret can be securely generated with the following 
+  # jwtSecret is the secret key you must use to sign and encrypt the JSON
+  # web token used to store user sessions. This secret key is
+  # critical for the security of Redpanda Console's authentication and
+  # authorization system. Use a long, complex key with a combination of
+  # numbers, letters, and special characters. While you must use a minimum of
+  # 10 characters, Redpanda recommends using more than 32
+  # characters. For additional security, use a different secret key for
+  # each environment. jwtSecret can be securely generated with the following
```
Comment (AnneLeeperRP) [inline docs/manage/security/console/authentication.mdx:47]:
```suggestion
  # numbers, letters, and special characters. The minimum number of
  # characters is 10, but Redpanda recommends using more than 32
  # characters. For additional security, use a different secret key for
  # each environment.
  
  jwtSecret can be securely generated with the following
```

File: docs/manage/security/console/authentication.mdx [hunk 3]
```diff
@@ -28,18 +37,18 @@ to add Google login support looks like this:
...
   # command: LC_ALL=C tr -dc '[:alnum:]' < /dev/random | head -c32
   #
-  # If you update this secret key, any users who are 
-  # already logged into Redpanda Console will be logged out and will have 
+  # If you update this secret key, any users who are
+  # already logged into Redpanda Console will be logged out and will have
```
Comment (AnneLeeperRP) [inline docs/manage/security/console/authentication.mdx:51]: "logged into" should be "logged in to" (and I suggested changing "update" to "change"):

```suggestion
  # If you change this secret key, any users who are
  # already logged in to Redpanda Console will be logged out and will have
```

File: docs/manage/security/console/azure-ad.mdx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+<head>
+    <meta name="title" content="Azure AD SSO Authentication in Redpanda Console | Redpanda Docs"/>
+    <meta name="description" content="Configure authentication with external identity providers such as Google, GitHub or Okta in Redpanda Console."/>
+</head>
+
+:::info
+This feature requires an [Enterprise license](../../../../get-started/licenses). To upgrade, contact [Redpanda sales](https://redpanda.com/try-redpanda?section=enterprise-cloud).
+:::
+
+Integrating Redpanda Console with Azure AD allows your users to use their Azure AD identities to sign-in to Console.
```
Comment (AnneLeeperRP) [inline docs/manage/security/console/azure-ad.mdx:14]:
```suggestion
Integrating Redpanda Console with Azure AD allows your users to use their Azure AD identities to sign in to Console.
```

File: docs/manage/security/console/azure-ad.mdx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+This guide assumes you already have an Azure AD account and permissions to create Applications within your directory.
+
+## Create an OpenID connect application
+
+TODO: Write text, possibly refer to: https://learn.microsoft.com/en-us/power-apps/maker/portals/configure/configure-openid-settings
+
+:::note
+Below configurations assume that you want to host Redpanda Console so that it would be accessible via
```
Comment (AnneLeeperRP) [inline docs/manage/security/console/azure-ad.mdx:22]:
```suggestion
The following configurations assume that you want to host Redpanda Console so that it would be accessible from
```
Another idea:
The following configurations assume that you want Redpanda Console to be accessible from

File: docs/manage/security/console/azure-ad.mdx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+`https://console.yourcompany.com`.
+:::
+
+- **Name:** Any descriptive name for your specific Console deployment (for example Console Analytics Prod)
```
Comment (AnneLeeperRP) [inline docs/manage/security/console/azure-ad.mdx:26]: For consistency, use the imperative:
```suggestion
- **Name:** Provide a descriptive name for your specific Console deployment (for example, Console Analytics Prod)
```

File: docs/manage/security/console/azure-ad.mdx [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    # in the next section.
+    # directory:
+    #   tenantId: ""
+```
+
+## RBAC Azure AD groups sync
+
+If you want to bind roles to Azure groups from your organization you have to provide the `tenantId` in the
+directory configuration and add API permissions to your client application. To retrieve the `tenantId` go to
+your registered application in the Azure Active Directory portal. The "Directory (tenant) ID" is listed
```
Comment (AnneLeeperRP) [inline docs/manage/security/console/azure-ad.mdx:89]:
```suggestion
To bind roles to Azure groups from your organization, provide the `tenantId` in the
directory configuration and add API permissions to your client application. To retrieve the `tenantId`, go to
your registered application in the Azure Active Directory portal. The "Directory (tenant) ID" is listed
```

File: docs/manage/security/console/azure-ad.mdx [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+along other configurations in the essentials section.
```
Comment (AnneLeeperRP) [inline docs/manage/security/console/azure-ad.mdx:90]:
```suggestion
To bind roles to Azure groups from your organization, provide the `tenantId` in the
directory configuration and add API permissions to your client application. To retrieve the `tenantId`, go to
your registered application in the Azure Active Directory portal. The "Directory (tenant) ID" is listed along with other configurations in the essentials section.
```

Comment (AnneLeeperRP) [inline docs/manage/security/console/azure-ad.mdx:90]:
```suggestion
To bind roles to Azure groups from your organization, provide the `tenantId` in the
directory configuration and add API permissions to your client application. To retrieve the `tenantId`, go to
your registered application in the Azure Active Directory portal. The "Directory (tenant) ID" is listed
along with other configurations in the essentials section.
```

File: docs/manage/security/console/authentication.mdx (hunk 2 above)
Comment (weeco) [reply docs/manage/security/console/authentication.mdx:47]: This comment is copy pasted for consistency across all IdPs. This had been reviewed for other pages before, thus I prefer to keep it as is or change everything if needed.

File: docs/manage/security/console/authentication.mdx (hunk 3 above)
Comment (weeco) [reply docs/manage/security/console/authentication.mdx:51]: This comment is copy pasted for consistency across all IdPs. This had been reviewed for other pages before, thus I prefer to keep it as is or change everything if needed.

File: docs/manage/security/console/azure-ad.mdx (hunk 4 above)
Comment (weeco) [reply docs/manage/security/console/azure-ad.mdx:14]: This comment is also copied and I think this dash had been removed in a previous PR. Whatever is the right version should go into all pages I assume?

File: docs/manage/security/console/keycloak.mdx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    # use Keycloak groups in your role bindings. 
+    # Described further in the next section.
+    # directory:
+    #   adminUser: ""
+    #   adminPassword: ""
+```
+
+## RBAC Keycloak groups sync
+
+If you want to bind roles to Keycloak groups you need to specifcy admin user credentials. These credentials
```
Comment (AnneLeeperRP) [inline docs/manage/security/console/keycloak.mdx:67]:
```suggestion
If you want to bind roles to Keycloak groups, you need to specify admin user credentials. These credentials
```

File: docs/manage/security/console/azure-ad.mdx (hunk 4 above)
Comment (Feediver1) [reply docs/manage/security/console/azure-ad.mdx]: Looks like we need to improve the process for docs reviewing these other source pages. Keeping mistakes/writing errors because they appear in other docs and we don't want them to be inconsistent is not a good approach. In other words, just because the IpD docs have errors does not mean ours should...

File: docs/manage/security/console/azure-ad.mdx (hunk 5 above)
Comment (Feediver1) [reply docs/manage/security/console/azure-ad.mdx]: A configuration cannot make an assumption. A configuration can be based on an assumption. So suggesting:

The following configurations are based on the assumption that you want to host Redpanda Console so that it is accessible from...

Comment (AnneLeeperRP) [reply docs/manage/security/console/azure-ad.mdx]: Shortened to:
The following configurations are based on hosting Redpanda Console so it is accessible from
`https://console.<your-company>.com`.

File: docs/manage/security/console/azure-ad.mdx (hunk 8 above)
Comment (AnneLeeperRP) [reply docs/manage/security/console/azure-ad.mdx]: Done.

File: docs/manage/security/console/azure-ad.mdx (hunk 4 above)
Comment (AnneLeeperRP) [reply docs/manage/security/console/azure-ad.mdx]: @Feediver1 
Martin is saying that he used the same wording in our docs for each IdP, so any mistakes were propagated to all our IdP documentation. I can file an issue to make all our SSO docs follow the same basic outline, and I can point out that this error needs to be fixed.

File: docs/manage/security/console/keycloak.mdx (hunk 9 above)
Comment (AnneLeeperRP) [reply docs/manage/security/console/keycloak.mdx]: Done.

File: docs/manage/security/console/authentication.mdx [hunk 10]
```diff
@@ -3,21 +3,30 @@ title: Authentication
...
 <head>
-    <meta name="title" content="SSO Authentication in Redpanda Console | Redpanda Docs"/>
-    <meta name="description" content="Configure authentication with external identity providers such as Google, GitHub or Okta in Redpanda Console."/>
+  <meta
+    name="title"
+    content="SSO Authentication in Redpanda Console | Redpanda Docs"
+  />
+  <meta
+    name="description"
+    content="Configure authentication with external identity providers such as Google, GitHub or Okta in Redpanda Console."
```
Comment (weeco) [inline docs/manage/security/console/authentication.mdx:12]: @AnneLeeperRP Should we update this as there are more providers now?

File: docs/manage/security/console/keycloak.mdx [hunk 11]
```diff
@@ -0,0 +1,140 @@
+---
+title: Keycloak SSO Setup
+---
+
+<head>
+    <meta name="title" content="Keycloak SSO Authentication in Redpanda Console | Redpanda Docs"/>
+    <meta name="description" content="Configure authentication with external identity providers such as Google, GitHub or Okta in Redpanda Console."/>
```
Comment (weeco) [inline docs/manage/security/console/keycloak.mdx:7]: Same question here if we should include Keycloak for this meta description

File: docs/manage/security/console/keycloak.mdx [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+This feature requires an [Enterprise license](../../../../get-started/licenses). To upgrade, contact [Redpanda sales](https://redpanda.com/try-redpanda?section=enterprise-cloud).
+:::
+
+By integrating Redpanda Console with Keycloak, your users can sign in to Redpanda Console using their Keycloak login credentials.
+
+## Prerequisites
+
+Install Keycloak and create a super admin user account, which has the necessary permissions for setting up the OAuth application.
+
+Create at least one realm for managing and authenticating a specific group of users.
```
Comment (weeco) [inline docs/manage/security/console/keycloak.mdx:20]: Keycloak has at least one realm by default (the master realm always exists)

File: docs/manage/security/console/keycloak.mdx [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+
+11. Copy the **Client secret**, which is automatically generated.
+
+:::info
+The `redpanda-console-config.yaml` file uses the client secret to establish communication with Keycloak.
+:::
+
+## Edit the console configuration file
+
+Edit the `redpanda-console-config.yaml` file, which is in `/etc/redpanda`, and incorporate the details from your client application.
```
Comment (weeco) [inline docs/manage/security/console/keycloak.mdx:68]: Same comment here again - it depends on how you're installing Redpanda Console.

File: docs/manage/security/console/keycloak.mdx [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+    # This is described further in the next section.
+    # directory:
+    #   adminUser: ""
+    #   adminPassword: ""
+```
+
+## RBAC Keycloak groups sync
+
+To bind roles to Keycloak groups, you need to specify admin user credentials. These credentials
+are used to resolve group memberships when communicating with Keycloak's API.
```
Comment (weeco) [inline docs/manage/security/console/keycloak.mdx:113]: Maybe we should mention here what admin user credentials => Those that can be used to login to the Keycloak admin panel too. Also that user should have administrative permissions on the realm the user is using to resolve the group memberships. I don't know the details of the exact permission set that is required, afaik this can also be customized in each Keycloak environment.

File: docs/manage/security/console/authentication.mdx (hunk 10 above)
Comment (AnneLeeperRP) [reply docs/manage/security/console/authentication.mdx:12]: Good point. Here are some questions to think about:
* How many IdPs will we support in the future? (Too many to list?)
* If we only mention some of them in this description (and we say "such as", which implies the list is not exhaustive), which ones deserve to be mentioned? The most commonly used ones?
* What about leaving out specify IdPs and just saying "Configure authentication with external identity providers in Redpanda Console"?

I think I like the last idea the best.

Comment (weeco) [reply docs/manage/security/console/authentication.mdx:12]: > How many IdPs will we support in the future? (Too many to list?)

Possibly we are already there.

> If we only mention some of them in this description (and we say "such as", which implies the list is not exhaustive), which ones deserve to be mentioned? The most commonly used ones?

Possibly. I think the listed three are the most commonly used ones. Maybe also AzureAD

> What about leaving out specify IdPs and just saying "Configure authentication with external identity providers in Redpanda Console"?

Also okay for me

File: docs/manage/security/console/azure-ad.mdx [hunk 15]
```diff
@@ -0,0 +1,169 @@
...
+
+When the client secret is generated successfully, it appears under the **Client secrets** tab with its description, expiration date, value, and secret ID.
+
+:::important
+Copy the client secret value into the `redpanda-console-config.yaml` file before you leave the **Certificates and secrets** page. After you navigate away from the page, only the first few characters of the client secret value are displayed and you won't be able to copy the value into the configuration file.
+:::
+
+## Edit the console configuration file
+
+Edit the `redpanda-console-config.yaml` file, which is in `/etc/redpanda`, and incorporate the details from your app registration.
```
[the remark this answers is not in the document]
Comment (weeco) [reply docs/manage/security/console/azure-ad.mdx:56]: Yes users would provide the configurations (not only Redpanda Console but also other configs) via the `values.yaml`

File: docs/manage/security/console/keycloak.mdx (hunk 12 above)
Comment (AnneLeeperRP) [reply docs/manage/security/console/keycloak.mdx:20]: What about saying "You can use the master realm, or you can create a realm for managing and authenticating a specific group of users."

That way, we're showing that users have a choice; and we're leaving the decision up to them as to whether they use the default (master) realm or not.

File: docs/manage/security/console/azure-ad.mdx (hunk 15 above)
[the remark this answers is not in the document]
Comment (AnneLeeperRP) [reply docs/manage/security/console/azure-ad.mdx:56]: What do you think of this rewrite:
>Edit the console configuration file associated with your deployment method and incorporate the details from your client application. For example, Kubernetes deployments use the `values.yaml` file. Linux deployments use the `redpanda-console-config.yaml` file, which is in `/etc/redpanda`.

I'm not sure if our users need that much guidance, but it seemed more complete to include that information. If you think this is unnecessary, what about:
>Edit the console configuration file associated with your deployment method and incorporate the details from your client application.

[the remark this answers is not in the document]
Comment (weeco) [reply docs/manage/security/console/azure-ad.mdx:56]: Both is fine for me.

File: docs/manage/security/console/keycloak.mdx [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+    A summary page opens.
+
+9. Click the **Credentials** tab.
+
+10. In **Client Authenticator**, select **Client Id and Secret**, then click **Save**.
+
+11. Copy the **Client secret**, which is automatically generated.
+
+:::info
+The `redpanda-console-config.yaml` file uses the client secret to establish communication with Keycloak.
```
[the remark this answers is not in the document]
Comment (AnneLeeperRP) [reply docs/manage/security/console/keycloak.mdx:67]: I thought it would be better to use the same terminology that Keycloak uses in their UI, so users know what we're talking about. People sometimes get confused if you use different language from the UI.

File: docs/manage/security/console/azure-ad.mdx (hunk 15 above)
[the remark this answers is not in the document]
Comment (AnneLeeperRP) [reply docs/manage/security/console/azure-ad.mdx:56]: I went with the first rewrite, which has more detail. 
@weeco  and @Feediver1 - Which do you prefer?

File: docs/manage/security/console/keycloak.mdx (hunk 14 above)
Comment (AnneLeeperRP) [reply docs/manage/security/console/keycloak.mdx:117]: @weeco 
If you think it's important to list the exact permission set (rather than saying "use the same admin credentials that are used for logging in to the admin panel"), we should find someone who knows the details.

Are you thinking that some users might need to create a separate admin user with a subset of admin panel permissions, to follow the principle of "least privileges"?

**Comment by Feediver1** [comment]
PR handing off to @JakeSCahill

File: docs/manage/security/console/azure-ad.mdx [hunk 17]
```diff
@@ -13,45 +14,24 @@ This feature requires an [Enterprise license](../../../../get-started/licenses).
...
 
-### Register your app
+- An OpenID Connect (OIDC) provider](https://learn.microsoft.com/en-us/power-apps/maker/portals/configure/configure-openid-settings) configured for your Azure AD account.
 
-To register your application, log in to the Azure Portal and provide the following information:
+## Create an OpenID connect application
 
-* **Display name** - Choose a descriptive name for your specific Redpanda Console deployment (for example Console Analytics Prod)
-* **Supported account types** - Select the types of accounts that can access the API (choose single tenant, multi-tenant, or personal accounts). If you are unsure, click "Help me choose...".
-* **Redirect URIs** Select Web as the platform and enter `https://console.<your-company>.com/login/callbacks/azure-ad` in the text box. Replace `<your-company>.com` with your company's domain name.
```
Comment (weeco) [inline docs/manage/security/console/azure-ad.mdx]: Where did that go? This is required information for the user. If they don't know the pattern for the callback url they would need to look through the code to find it and that is closed source.

File: docs/manage/security/console/azure-ad.mdx [hunk 18]
```diff
@@ -13,45 +14,24 @@ This feature requires an [Enterprise license](../../../../get-started/licenses).
...
-2. On the **Certificates & secrets** page, select the **Client secrets** tab.
-
-3. Click **New client secret**.
-
-4. In the **Add a client secret** pane, enter a description that identifies the app associated with the client secret.
-
-5. Choose a time period that determines when the client secret will expire; for example, 180 days.
-
-When the client secret is generated successfully, it appears under the **Client secrets** tab with its description, expiration date, value, and secret ID.
-
```
Comment (weeco) [inline docs/manage/security/console/azure-ad.mdx]: Why is this removed?

File: docs/manage/security/console/keycloak.mdx [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+
+## Prerequisites
+
+You must:
+
+- Install Keycloak and create a super admin user account, which has the necessary permissions for setting up the OAuth application. For details, see the [Keyclock documentation](https://www.keycloak.org/guides).
+- Create an OpenID Connect (OIDC) client. For details, see the [Keyclock documentation](https://www.keycloak.org/docs/latest/server_admin/index.html#proc-creating-oidc-client_server_administration_guide).
+
+  :::note
+  - Set the valid redirect URIs to the domain where Redpanda Console is hosted followed by the `/login/callbacks/keycloak` path. For example, `https://console.<your-company>.com/login/callbacks/keycloak`
```
Comment (JakeSCahill) [inline docs/manage/security/console/keycloak.mdx:25]:
```suggestion
  - Set the valid redirect URIs to the domain where Redpanda Console is hosted followed by the `/login/callbacks/keycloak` path. For example, `https://console.<your-company>.com/login/callbacks/keycloak`.
```

File: docs/manage/security/console/azure-ad.mdx [hunk 20]
```diff
@@ -0,0 +1,149 @@
...
+## Prerequisites
+
+You must have:
+
+- An [Azure AD account](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/sign-up-organization) and permissions to create applications within your directory.
+
+- An OpenID Connect (OIDC) provider](https://learn.microsoft.com/en-us/power-apps/maker/portals/configure/configure-openid-settings) configured for your Azure AD account.
+
+  :::note
+  Set the redirect URI to the domain where Redpanda Console is hosted followed by the `/login/callbacks/azure-ad` path. For example, `https://console.<your-company>.com/login/callbacks/azure-ad`
```
Comment (JakeSCahill) [inline docs/manage/security/console/azure-ad.mdx:26]:
```suggestion
  Set the redirect URI to the domain where Redpanda Console is hosted followed by the `/login/callbacks/azure-ad` path. For example, `https://console.<your-company>.com/login/callbacks/azure-ad`.
```

File: docs/manage/security/console/azure-ad.mdx [hunk 21]
```diff
@@ -13,45 +14,24 @@ This feature requires an [Enterprise license](../../../../get-started/licenses).
...
+## Prerequisites
 
-## Create an OpenID connect application
+You must have:
 
-In the Azure Portal, register your app and generate a client secret.
+- An [Azure AD account](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/sign-up-organization) and permissions to create applications within your directory.
 
-### Register your app
+- An OpenID Connect (OIDC) provider](https://learn.microsoft.com/en-us/power-apps/maker/portals/configure/configure-openid-settings) configured for your Azure AD account.
```
[the remark this answers is not in the document]
Comment (JakeSCahill) [reply docs/manage/security/console/azure-ad.mdx:23]: I've added a note below with the requirements for the redirect URI.

File: docs/manage/security/console/azure-ad.mdx (hunk 17 above)
Comment (JakeSCahill) [reply docs/manage/security/console/azure-ad.mdx]: I've kept the information about the redirect URI.

I've removed steps that document third-party tools because it's a best practice. We cannot guarantee that any steps we provide for third-party tools remain up to date.

File: docs/manage/security/console/azure-ad.mdx (hunk 18 above)
Comment (JakeSCahill) [reply docs/manage/security/console/azure-ad.mdx]: I've removed steps that document third-party tools because it's a best practice. We cannot guarantee that any steps we provide for third-party tools remain up to date.

Instead, we should provide links and any Redpanda-specific input requirements.

File: docs/manage/security/console/keycloak.mdx [hunk 22]
```diff
@@ -15,60 +16,22 @@ By integrating Redpanda Console with Keycloak, your users can sign in to Redpand
...
-9. Click the **Credentials** tab.
-
-10. In **Client Authenticator**, select **Client Id and Secret**, then click **Save**.
-
-11. Copy the **Client secret**, which is automatically generated.
-
-:::info
-The console configuration file uses the client secret, which is basically a password, to establish communication with Keycloak.
-:::
-
```
[the remark this answers is not in the document]
Comment (JakeSCahill) [reply docs/manage/security/console/keycloak.mdx]: I've removed steps that document third-party tools because it's a best practice. We cannot guarantee that any steps we provide for third-party tools remain up to date.

File: docs/manage/security/console/azure-ad.mdx [hunk 23]
```diff
@@ -0,0 +1,149 @@
...
+
+By integrating Redpanda Console with Azure AD, your users can sign in to Redpanda Console using their Azure AD login credentials.
+
+## Prerequisites
+
+You must have:
+
+- An [Azure AD account](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/sign-up-organization) and permissions to create applications within your directory.
+
+- An OpenID Connect (OIDC) provider](https://learn.microsoft.com/en-us/power-apps/maker/portals/configure/configure-openid-settings) configured for your Azure AD account.
```
Comment (Feediver1) [inline docs/manage/security/console/azure-ad.mdx:23]:
```suggestion
- An [OpenID Connect (OIDC) provider](https://learn.microsoft.com/en-us/power-apps/maker/portals/configure/configure-openid-settings) configured for your Azure AD account.
```

Comment (Feediver1) [reply docs/manage/security/console/azure-ad.mdx:23]: ^^ Link wan't rendering correctly

**Review by Feediver1 — APPROVED** [review]
Be sure to change those "Keyclock" typos.
